### ISSUE TITLE
sls create: Install missing plugins automatically

### ISSUE THREAD

**Issue by a-h** [serverless/serverless#7938]
### Use case description

The nice thing about Serverless Framework is the dev experience. `sls create` / `sls deploy`. 

But if the author has used a plugin, and you try to deploy it, you get something like this:

```
Serverless plugin "serverless-step-functions" not found. Make sure it's installed and listed in the "plugins" section of your serverless config file.
```

Then you have to Google to find out you need to run something like:

```
sls plugin install -n serverless-step-functions
```

### Proposed solution

If the plugin isn't there, run the command automatically for the user to install the missing plugins.

**Comment by medikoo** [comment]
@a-h that's very good point. Thanks for posting. We'll look into, how we can improve that.

I believe after putting a project in place, if it doesn't have `package.json` we should read list of eventual plugins and install all of them via `sls plugin install -n [plugin-name]`

**Comment by a-h** [comment]
Nice. In this case, the project does have a `package.json`, which was created by `sls create`, then the `devDependenies` section added when the `serverless-step-functions` plugin was added.

```
{
  "name": "ddbimport",
  "description": "",
  "version": "0.1.0",
  "dependencies": {},
  "devDependencies": {
    "serverless-step-functions": "^2.21.1"
  }
}
```

The workflow here is that if one developer creates a Serverless project, adds a plugin, uploads it to Github, and then another developer clones the repo and tries to execute `sls deploy`, they'll get this "plugin not found" error.

In my example case, the project isn't a JavaScript project, all the Lambda Functions are written in Go, so the developer wouldn't need to run `npm install` to build the project prior to deployment. This makes it much more likely a developer will experience this usability friction if they're building anything other than a JavaScript or TypeScript project.

**Comment by medikoo** [comment]
> Nice. In this case, the project does have a `package.json`

In such case I believe `npm install` is run by `sls init`

> tries to execute sls deploy, they'll get this "plugin not found" error.

`sls deploy` just deploys poject, and it assumes project is already installed. I don't think any installation steps should be hardcoded into it, as it'll slow down the deploy and in general may introduce unexpected side effects.

**Comment by a-h** [comment]
I tested this out:

```
sls create --template=aws-go
```

No `package.json` is created.

```
sls plugin install -n serverless-step-functions
```

A `package.json` is created...

```
Serverless: Creating an empty package.json file in your service directory
Serverless: Installing plugin "serverless-step-functions@latest" (this might take a few seconds...)
Serverless: Successfully installed "serverless-step-functions@latest"
```

Time to deploy!

`sls deploy`

It works. However, if you then commit it to `git` and get someone to download the project, and run `make deploy` as per the Makefile, you get the error.

It's really just a usability thing.

**Comment by medikoo** [comment]
> It works. However, if you then commit it to git and get someone to download the project

By _download_ you mean simply doing `git clone`, or installation via `sls init` ?

**Comment by a-h** [comment]
Yes, sorry, I mean by executing a `git clone` to clone the repo, or by downloading the zip of the code, no use of `sls init` etc.

**Comment by medikoo** [comment]
@a-h thanks for clarification, still I'm getting a bit lost. In main description you've specifically referenced `sls init` and in further comments you further describe the case not using `sls init`.

Can you explain what exactly is your report about

**Comment by a-h** [comment]
Sorry, it was a typo in the initial comment. I should have written `sls create`. I got mixed up because so many tools start with `init`, e.g. `git init`, `go mod init`, `npm init`.

**Comment by medikoo** [comment]
@a-h thanks for clarification, that significantly changes the scope, as `sls init` is a command we've implemented recently which follows up with `npm install` (if `package.json` is part of a template).

I take then, this is a request to install missing plugins on `sls create` (so issue `npm install` if `package.json` is there) Note that in Serverless ecosystem it is assumed that plugins are published to npm, and it's the only package manager that internally is supported by the CLI

**Comment by francopiccolo** [comment]
> > Nice. In this case, the project does have a `package.json`
> 
> In such case I believe `npm install` is run by `sls init`
> 
> > tries to execute sls deploy, they'll get this "plugin not found" error.
> 
> `sls deploy` just deploys poject, and it assumes project is already installed. I don't think any installation steps should be hardcoded into it, as it'll slow down the deploy and in general may introduce unexpected side effects.

`sls init` doesn't install packages from the `package.json` file, it would be great if it did or creating the `sls create` suggested by @a-h

**Comment by medikoo** [comment]
@francopiccolo this isssue is about `sls create` if you have issue report or feature request that relates to `sls init` please open a new issue

**Comment by medikoo** [comment]
Closing as it can be considered as addressed with https://github.com/serverless/serverless/pull/9483

For project setup we're moving from `sls create` to `sls`, and when setting up the project with `sls` dependencies are installed for copied template (released with v2.42.0)
